Implement Multilingual Translation Functionality

MouniaBelkheir opened the pull request:
## Pull Request: Implement Multilingual Translation Functionality
### Description:
This pull request introduces comprehensive multilingual translation functionality to the website. The implementation covers English, French, and Arabic languages, enhancing the overall user experience for a broader audience. Here's an overview of the changes:

- Identified and listed translatable content across the website.
- Refactored hardcoded strings to dynamic and translatable elements.
- Integrated the i18next translation library for efficient language handling.
- Created separate translation files for English, French, and Arabic.
- Configured dynamic loading of translation files based on user language preference.
- Conducted extensive testing for accurate translations and proper rendering.
- Ensured UI/UX, cross-browser, and accessibility testing for all supported languages.

**Testing Status:**

All tests, including translation correctness, UI/UX, cross-browser, and accessibility, have passed successfully.

**Screenshots**

### On Large Screens

![image](https://github.com/202306-NEA-DZ-FEW/StudentStore/assets/137924594/a5a9cdaf-a9b8-4034-bc35-3b2f6bb1f5fb)

![image](https://github.com/202306-NEA-DZ-FEW/StudentStore/assets/137924594/3b2956e0-5222-401d-a3eb-a4e6b8ccfcea)

![image](https://github.com/202306-NEA-DZ-FEW/StudentStore/assets/137924594/e08c8c79-5f25-4e5f-a2f7-ff27a5d68336)

### On Tablets:

<img width="617" alt="image" src="https://github.com/202306-NEA-DZ-FEW/StudentStore/assets/137924594/950376e4-f2ac-4dbc-bf45-40917b2ebf2f">

<img width="599" alt="image" src="https://github.com/202306-NEA-DZ-FEW/StudentStore/assets/137924594/de8dd481-ac53-4d6a-bf65-61a64744022a">

### On Small Screens

<img width="194" alt="image" src="https://github.com/202306-NEA-DZ-FEW/StudentStore/assets/137924594/5a94fa00-8075-431a-b863-71b1effbe168">

<img width="204" alt="image" src="https://github.com/202306-NEA-DZ-FEW/StudentStore/assets/137924594/6e79fc8c-4f5c-4b34-86cd-dde774535631">


# Related Issue

- Resolve #127

MouniaBelkheir committed:
feat(navbar & footer & home page): add arabic & french translation for navbar & footer & home page

MouniaBelkheir committed:
fix(home page): resolve merge conflict

MouniaBelkheir committed:
feat(translation for latest products): add arabic & french translation to the Latest products

MouniaBelkheir committed:
feat(event card): add translation to event card

add translation to up comming events

MouniaBelkheir committed:
perf(home page anddonation page): rephrase translation

Improve arabic and french translation to the home page and donation page and adjusting punctuations

#127

MouniaBelkheir committed:
perf(side bar): improve appearance

improve text appearance via small screens view in the side bar of the nav ar

MouniaBelkheir committed:
style: prettier test

MouniaBelkheir committed:
style(prettier test 2): prettier 2

MouniaBelkheir committed:
refactor(locale): add translation to the footer

samiba6 reviewed (approved):
Well done @MouniaBelkheir the translation is accurate and it's working smoothly

MouniaBelkheir committed:
perf(footer): adjust footer's styling

improved the styling of the footer and tested how does it look  while toggling between the three
languages, on diffrent  screen sizes

#127

MouniaBelkheir committed:
Merge branch 'develop' of github.com:202306-NEA-DZ-FEW/StudentStore into 127-translation-implementation-functionality

MouniaBelkheir committed:
perf(navbar): adjust the navbar's styling

improved the navbar styling and fixed its issues when toggling between languages

#127

KatiaGhezali reviewed (approved):
Good job Mounia, thank you.
